drivers/hmis/lib/hmis_data_cleanup/migrate_units_to_unit_groups_20250828.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+class HmisDataCleanup::MigrateUnitsToUnitGroups20250828
+  def initialize
+  end
+
+  def perform
+    Hmis::UnitGroup.transaction do
+      # First, fix existing unit groups that don't have unit_type set
+      add_unit_type_to_unit_groups
+
+      # Then, create unit groups for projects that don't have any yet
+      create_unit_groups_for_projects
+
+      puts 'Migration complete!'
+
+      # raise ActiveRecord::Rollback # uncomment to test rollback
+    end
+  end
+
+  private
+
+  def add_unit_type_to_unit_groups
+    puts 'Fixing existing unit groups without unit_type'
+
+    # Find all unit groups that don't have a unit_type set
+    unit_groups_without_type = Hmis::UnitGroup.where(unit_type: nil).includes(:units)
+
+    puts "Found #{unit_groups_without_type.count} unit groups without unit_type"
+
+    unit_groups_updated = 0
+
+    unit_groups_without_type.find_each do |unit_group|
+      units = unit_group.units.where.not(unit_type: nil)
+      next if units.empty?
+
+      # Get all distinct unit types for units in this unit group
+      unit_type_ids = units.distinct.pluck(:unit_type_id)
+
+      raise "Unit group migration - unexpected error! Unit group #{unit_group.id} (#{unit_group.name}) has units with different unit types." if unit_type_ids.size > 1
+
+      # All units have the same unit_type, set it on the unit group
+      unit_type = Hmis::UnitType.find(unit_type_ids.first)
+      unit_group.update!(unit_type: unit_type)
+      unit_groups_updated += 1
+    end
+
+    puts "Updated #{unit_groups_updated} unit groups"
+  end
+
+  def create_unit_groups_for_projects
+    puts 'Creating unit groups for projects that have units without groups'
+
+    # Find all project/unit_type combinations that have units but no unit groups
+    projects_without_unit_groups = Hmis::Hud::Project.
+      joins(:units).
+      left_joins(:unit_groups).
+      where(hmis_unit_groups: { id: nil }).
+      distinct.
+      includes(:units)
```

Comment:
This is fine, I usually find `Hmis::Hud::Project.where.missing(:unit_groups)` helpful with this kind of query since it's a bit easier to read.

Can we add a `hmis` scope to ensure we are only dealing with projects in the HMIS data source? (Others shouldn't have Units,  but just to be safe)